.github/workflows/ci.yml

```diff
@@ -0,0 +1,59 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build-and-test:
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Restore tools
+        run: dotnet tool restore
+        working-directory: ./Src
+      - name: Build templates
+        # TargetFramework specified so this only runs once rather than per framework
+        run: dotnet msbuild -t:TextTemplateTransform -p:TargetFramework=NA
+        working-directory: ./Src
+      - name: Debug Build
+        run: dotnet build --configuration=Debug -p:TreatWarningsAsErrors=true
+        working-directory: ./Src
+      - name: Release Build
+        run: dotnet build --configuration=Release -p:TreatWarningsAsErrors=true
+        working-directory: ./Src
+      - name: CPU Tests
+        run: dotnet test --configuration=Release
+        working-directory: ./Src/ILGPU.Tests.CPU
+      - name: Upload bin directory
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ runner.os }}-bin
+          path: Bin
+
+  nuget-package:
+    runs-on: windows-latest
+    needs: build-and-test
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Download Windows binaries
+        uses: actions/download-artifact@v2
+        with:
+          name: Windows-bin
+          path: Bin
+      - name: Set up NuGet
+        uses: nuget/setup-nuget@v1
+        with:
+          nuget-version: '5.x'
+      - name: Create NuGet package
+        run: nuget pack ILGPU.nuspec
+        working-directory: .nuget
```

Comment:
I wonder whether we should not create the nuget package directly inside the previous job. In that case, there would be no need to upload the bin directory as an artifact. The resulting artifact would only be the nuget package.

In the future, once we can run tests on multiple platforms, we might want to have a separate job for publishing the nuget package when all tests succeed and the CI is running for a tag like `v*`.

Reply:
Sounds reasonable; furthermore, this should simplify the whole internal deployment/test process.

Reply:
Yeah I separated them out thinking it would be cleaner if we built on multiple OSs and then created a NuGet package just from the Windows build, but we could still do that by conditionally running the NuGet package step and it would be simpler.